### ISSUE TITLE
Node builds don't work?

### ISSUE THREAD

**Issue by melMass** [honojs/vite-plugins#194]
Hi,

I'm trying these plugins mostly to be able to easily bundle assets but I'm confused about a few things after playing around with it.

I was surprised by the routes added by the vite plugin to serve `/static` & `/assets`, this should probably be explained somewhere.
I debugged it for a long time before understanding that (since I was also declaring a `/static` route myself.)

I followed the vite-jsx example and even tried adding a few more routes, it all works fine in dev (using your vite plugin too), but running the built server in node errors out, subroutes work but the `/` route (where I serve the client from the sample) returns this :
<img width="560" alt="image" src="https://github.com/user-attachments/assets/a72be8ae-df1d-4a60-b892-58392b1414f9">

<details><summary>as text</summary>

```sh
TypeError: Cannot read properties of undefined (reading 'call')
    at file:///Users/melmassadian/dev/melmass/api/dist/index.js:1938:57
    at Array.<anonymous> (file:///Users/melmassadian/dev/melmass/api/dist/index.js:3133:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async responseViaResponseObject (file:///Users/melmassadian/dev/melmass/api/dist/ind
ex.js:3552:10)
TypeError: Cannot read properties of undefined (reading 'call')
    at file:///Users/melmassadian/dev/melmass/api/dist/index.js:1938:57
    at Array.<anonymous> (file:///Users/melmassadian/dev/melmass/api/dist/index.js:3133:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async responseViaResponseObject (file:///Users/melmassadian/dev/melmass/api/dist/ind
ex.js:3552:10)
Error: Context is not finalized. Did you forget to return a Response object or `await next(
)`?
    at file:///Users/melmassadian/dev/melmass/api/dist/index.js:1974:12
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async responseViaResponseObject (file:///Users/melmassadian/dev/melmass/api/dist/ind
ex.js:3552:10)
Error: Context is not finalized. Did you forget to return a Response object or `await next(
)`?
    at file:///Users/melmassadian/dev/melmass/api/dist/index.js:1974:12
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async responseViaResponseObject (file:///Users/melmassadian/dev/melmass/api/dist/ind
ex.js:3552:10)

^CError: nu::shell::terminated_by_signal

  × External command was terminated by a signal
   ╭─[entry #48:1:1]
 1 │ node dist/index.js
   · ──┬─
   ·   ╰── terminated by SIGINT (2)
   ╰────
   ```

</details>

This also happens if I remove all other routes/middleware...

Any suggestions are welcome, I really love the modularity of Hono

**Comment by yusukebe** [comment]
Hi @melMass 

Can you share instructions on how to reproduce it?

**Comment by melMass** [comment]
Sure I'll make a repro now!

**Comment by melMass** [comment]
Here is the repro: https://github.com/melMass/reproduction/tree/js/hono-vite-build


Not sure if you can see the action result here: https://github.com/melMass/reproduction/actions/runs/12173543142/job/33954069297

The relevant part is the `Start server and test`:

```sh
Page loaded with status: 200
Page content: <html lang="en"><head><meta charset="utf-8"><meta content="width=device-width, initial-scale=1" name="viewport"><link rel="stylesheet" href="/static/style.css"><script type="module" src="/static/client.js"></script></head><body><div id="root"></div></body></html>
=== Server Logs ===
TypeError: Cannot read properties of undefined (reading 'call')
    at file:///home/runner/work/reproduction/reproduction/dist/index.js:[14](https://github.com/melMass/reproduction/actions/runs/12173543142/job/33954069297#step:7:15)97:60
    at Array.<anonymous> (file:///home/runner/work/reproduction/reproduction/dist/index.js:2420:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async responseViaResponseObject (file:///home/runner/work/reproduction/reproduction/dist/index.js:2792:13)
TypeError: Cannot read properties of undefined (reading 'call')
    at file:///home/runner/work/reproduction/reproduction/dist/index.js:1497:60
    at Array.<anonymous> (file:///home/runner/work/reproduction/reproduction/dist/index.js:2420:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async responseViaResponseObject (file:///home/runner/work/reproduction/reproduction/dist/index.js:2792:13)
```

It took more time than expected to make the CI log the error since as you can see the client receives a 200 even though the js failed

**Comment by yusukebe** [comment]
@melMass 

Thanks! It's a bug and I found the cause. I'll fix it.

**Comment by yusukebe** [comment]
@melMass 

This issue may be fixed with `@hono/vite-build@1.1.1`. Try it!

**Comment by melMass** [comment]
Nice thanks, I just updated and reran the action and it seems to work!
I'll test once I'm back at the computer locally (no errors in the server log)
